src/main/java/at/hannibal2/skyhanni/features/misc/chocolatefactory/HoppityEggsLocations.kt

```diff
@@ -0,0 +1,221 @@
+                eggLocation,
+                LorenzColor.GREEN.toColor(),
+                seeThroughBlocks = true,
+            )
+            event.drawDynamicText(eggLocation.add(y = 1), "§aEgg", 1.5)
+        }
+    }
+
+    fun eggFound() {
+        resetData()
+    }
+
+    @SubscribeEvent
+    fun onReceiveParticle(event: ReceiveParticleEvent) {
+        if (!isEnabled()) return
+        if (!hasLocatorInInventory()) return
+        if (!event.isVillagerParticle() && !event.isEnchantmentParticle()) return
+
+        val lastParticlePosition = lastParticlePosition ?: run {
+            this.lastParticlePosition = event.location
+            return
+        }
+        if (lastParticlePosition == event.location) {
+            validParticleLocations.add(event.location)
+            ticksSinceLastParticleFound = 0
+        }
+        this.lastParticlePosition = null
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+        if (validParticleLocations.isEmpty()) return
+        ticksSinceLastParticleFound++
+
+        if (ticksSinceLastParticleFound < 6) return
+
+        calculateEggPosition()
+
+        ticksSinceLastParticleFound = 0
+        validParticleLocations.clear()
+    }
+
+    private fun calculateEggPosition() {
+        val islandEggsLocations = getCurrentIslandEggLocations() ?: return
+        val listSize = validParticleLocations.size
+        if (listSize < 5) return
+
+        val secondPoint = validParticleLocations.removeLast()
+        val firstPoint = validParticleLocations.removeLast()
+
+        val xDiff = secondPoint.x - firstPoint.x
+        val yDiff = secondPoint.y - firstPoint.y
+        val zDiff = secondPoint.z - firstPoint.z
+
+        firstPos = firstPoint.roundLocationToBlock()
+
+        secondPos = LorenzVec(
+            secondPoint.x + xDiff * 1000,
+            secondPoint.y + yDiff * 1000,
+            secondPoint.z + zDiff * 1000
+        ).roundLocationToBlock()
+
+        val worldBoundingBox = getWorldBoundingBox(islandEggsLocations)
+
+        val boundingBox = AxisAlignedBB(
+            firstPos.x, firstPos.y, firstPos.z,
+            secondPos.x, secondPos.y, secondPos.z
+        ).expand(5.0).clampTo(worldBoundingBox)
+
+        possibleEggLocations.clear()
+
+        // todo allow more leeway for further points later
+        // need to test while is hoppity event. Works well enough for now
+        val a = islandEggsLocations.filter {
+            boundingBox.isInside(it) && it.distanceToLine(firstPos, secondPos) < 200.0
+        }
+
+        possibleEggLocations.addAll(a.sortedBy {
+            it.distanceToLine(firstPos, secondPos)
+        })
+
+        drawLocations = true
+    }
+
+    private fun getWorldBoundingBox(islandEggs: List<LorenzVec>): AxisAlignedBB {
+        var minX = 10000.0
+        var minY = 10000.0
+        var minZ = 10000.0
+        var maxX = -10000.0
+        var maxY = -10000.0
+        var maxZ = -10000.0
```

Comment:
Double.positiveInfinit and Double.negativInfinit.
Works for any value. Since this way you lock it to 10000 (which is reasonable) but it could be bigger.

Reply:
i dont think hypixel is making islands bigger than 10k x 10k anytime soon